packages/react-scripts/config/webpack.config.js

```diff
@@ -718,33 +717,49 @@ module.exports = function (webpackEnv) {
       // TypeScript type checking
       useTypeScript &&
         new ForkTsCheckerWebpackPlugin({
-          typescript: resolve.sync('typescript', {
-            basedir: paths.appNodeModules,
-          }),
           async: isEnvDevelopment,
-          checkSyntacticErrors: true,
-          resolveModuleNameModule: process.versions.pnp
-            ? `${__dirname}/pnpTs.js`
-            : undefined,
-          resolveTypeReferenceDirectiveModule: process.versions.pnp
-            ? `${__dirname}/pnpTs.js`
-            : undefined,
-          tsconfig: paths.appTsConfig,
-          reportFiles: [
+          typescript: {
+            typescriptPath: resolve.sync('typescript', {
+              basedir: paths.appNodeModules,
+            }),
+            configOverwrite: {
+              compilerOptions: {
+                sourceMap: isEnvProduction
+                  ? shouldUseSourceMap
+                  : isEnvDevelopment,
+                skipLibCheck: true,
+                inlineSourceMap: false,
+                declarationMap: false,
+                noEmit: true,
+                incremental: true,
+              },
+            },
+            context: paths.appPath,
+            diagnosticOptions: {
+              syntactic: true,
+            },
+            mode: 'write-references',
+            // profile: true,
+          },
+          issue: {
             // This one is specifically to match during CI tests,
             // as micromatch doesn't match
             // '../cra-template-typescript/template/src/App.tsx'
             // otherwise.
-            '../**/src/**/*.{ts,tsx}',
-            '**/src/**/*.{ts,tsx}',
-            '!**/src/**/__tests__/**',
-            '!**/src/**/?(*.)(spec|test).*',
-            '!**/src/setupProxy.*',
-            '!**/src/setupTests.*',
-          ],
-          silent: true,
-          // The formatter is invoked directly in WebpackDevServerUtils during development
-          formatter: isEnvProduction ? typescriptFormatter : undefined,
+            include: [
+              '../**/src/**/*.{ts,tsx}',
+              '**/src/**/*.{ts,tsx}',
+            ].map(file => ({ file })),
+            exclude: [
+              '**/src/**/__tests__/**',
+              '**/src/**/?(*.)(spec|test).*',
+              '**/src/setupProxy.*',
+              '**/src/setupTests.*',
+            ].map(file => ({ file })),
```

Comment:
@piotr-oles I tried reading the documentation in fork-ts-checker-webpack-plugin, but I ended up looking at the source code instead - not sure if this is correct in terms of use case not sure if the minimatch for files support pattern matching?

https://github.com/TypeStrong/fork-ts-checker-webpack-plugin#issues-options array of string / patterns are not supported anymore according to the configuration validation error I was faced with.

Reply:
on v6.0.5 now

Reply:
Yes, the documentation of this configuration option is pretty bad :/ I should improve it... Could you create an issue on the plugin repo for tracking bugs that you found?

Reply:
@piotr-oles I've added an issue for tracking https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/548

I didn't encounter other issues what I know of - it seems like the plugin is closer integrated with webpack now meaning we can remove some code in this project.

I'll remove this - let me know if I missed out on anything :)